app/src/main/java/se/nullable/flickboard/ui/voice/VoiceSwitcher.kt

```diff
@@ -0,0 +1,27 @@
+package se.nullable.flickboard.ui.voice
+
+import android.view.inputmethod.InputMethodInfo
+import android.view.inputmethod.InputMethodManager
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalContext
+import androidx.core.content.getSystemService
+
+@Composable
+fun getAvailableVoiceInputMethod() : String? {
+    val context = LocalContext.current
+    val inputManager =
+        remember(context) { context.getSystemService<InputMethodManager>() } ?: return null
```

Comment:
Not 100% sure, but I _think_ we might as well just pass in the IMM since we're only using this from the `KeyboardService` anyway.

Reply:
Based on how it cleaned things up I think this was the right call? Since the wrapping function still isn't Composable I'm not currently using `remember` for the fetched IMM, which may have implications I'm not aware of; but otherwise this is pretty tidy and is still working on my device. 

I still need to figure out if a Snackbar is a sensible way to surface an error message (and in which context to do so) but I think that's the last remaining obstacle.